src/components/SearchBlock.tsx

```diff
@@ -91,54 +135,149 @@ export const SearchBlock = ({ versionId, filter }: SearchBlockProps): JSX.Elemen
             }
             {errorMessage !== '' && <p className='os-search-error-message'>{errorMessage}</p>}
           </Form>
         )}
       </Formik>
       {searchResults !== undefined && searchResults.hits.total.value !== 0 &&
-        <div className='os-search-results-container'>
-          <p>Total search results: {searchResults.hits.total.value}</p>
-          <p>Total search results displayed: {searchResults.hits.hits.length}</p>
-          <ul className='os-search-results-list'>
-            {searchResults.hits.hits.map((hit) => (
-              <li className='os-search-results-list-item' key={hit._id}>
-                <div>
-                  <h3>Location</h3>
-                  <p>{hit._source.section}</p>
-                  <p>{hit._source.activity_name}</p>
-                  <p>{hit._source.lesson_page !== '' && hit._source.lesson_page}</p>
+      <>
+        <div className='os-search-results-count-container'>
+          <h3 className='os-search-heading'>Search Results</h3>
+          <div>
+            <p className='os-search-magnifying-glass os-search-results-text'>
+              Displaying {teacherContentOnly ? totalHitsDisplayed : searchResults.hits.hits.length} of out {searchResults.hits.total.value} results for <span className='os-raise-text-bold'>{searchTerm}</span>
+            </p>
+            {filter === undefined &&
+              <div className='os-raise-d-flex-nowrap os-raise-justify-content-evenly os-search-teacher-content-toggle-container'>
+                <p className='os-raise-mb-0 os-search-results-text'>Teacher Content Only</p>
+                <div className='os-raise-bootstrap'>
+                  <div className="form-check form-switch">
+                    <input
+                      className="form-check-input"
+                      type="checkbox"
+                      id="flexSwitchCheckDefault"
+                      onChange={() => {
+                        setTeacherContentOnly(!teacherContentOnly)
+                        calculateTotalTeacherOnlyHits(searchResults)
+                      }}
+                    />
+                  </div>
+                </div>
+              </div>
+            }
+          </div>
+        </div>
+        {groupedHits !== undefined &&
+          <div className='os-search-results-container'>
+          {teacherContentOnly
+            ? <div className='os-raise-bootstrap'>
+                <div className="accordion" id="teacherContentAccordion">
+                  {Object.keys(groupedHits).sort().map((unitName) => {
+                    return (
+                      <div className="accordion-item" key={unitName}>
+                        <h3 className="accordion-header">
+                          <button
+                            className="accordion-button collapsed os-raise-text-bold"
+                            type="button" data-bs-toggle="collapse"
+                            data-bs-target={`#${unitName.slice(0, 6).replace(/\s/g, '')}`}
```

Comment:
Belated thought, but if we end up going down this path, perhaps using `unitName` at all doesn't buy anything relative to just:

```javascript
const uniqueId = `auto-${uuidv4()}`
```
:point_up: would avoid any possibility of a unit name potentially breaking things, and I think it would work okay :thinking: .

Reply:
Since we are calling `uuidv4()` in `key` I think the value for `key` changes each time we click the slider. Because the `key` is different each time, I think this causes a recreation of the applicable elements. Is that understanding correct?

Reply:
Do you mean the `key` for the hit? That is `hit.__id` so already guaranteed to be unique and unrelated, no?